tests/unit/test_cache.cpp

```diff
@@ -0,0 +1,54 @@
+#include "gtest.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+#include "../../src/util/rmalloc.h"
+#include "../../src/util/cache/cache.h"
+#include "../../src/execution_plan/execution_plan.h"
+#ifdef __cplusplus
+}
+#endif
+
+class CacheTest:
+	public ::testing::Test {
+  protected:
+	static void SetUpTestCase() {// Use the malloc family for allocations
+		Alloc_Reset();
+	}
+};
+
+TEST_F(CacheTest, ExecutionPlanCache) {
+	Cache *cache = Cache_New(3, (listValueFreeFunc)ExecutionPlan_Free);
```

Comment:
`listValueFreeFunc` should be renamed, if this function is used to free the cached item it should be named accordingly without revealing the underline implementation.